full-node/db/sov-schema-db/src/lib.rs

```diff
@@ -292,58 +62,101 @@ impl SchemaBatch {
+        #[cfg(feature = "std")]
+        let _timer = metrics::SCHEMADB_BATCH_PUT_LATENCY_SECONDS
             .with_label_values(&["unknown"])
             .start_timer();
+
         let key = key.encode_key()?;
         let value = value.encode_value()?;
+
+        #[cfg(feature = "std")]
         self.rows
             .lock()
             .expect("Lock must not be poisoned")
             .entry(S::COLUMN_FAMILY_NAME)
             .or_default()
             .push(WriteOp::Value { key, value });
 
+        #[cfg(not(feature = "std"))]
+        self.rows
+            .lock()
+            .entry(S::COLUMN_FAMILY_NAME)
+            .or_default()
+            .push(WriteOp::Value { key, value });
+
         Ok(())
     }
 
     /// Adds a delete operation to the batch.
     pub fn delete<S: Schema>(&self, key: &impl KeyCodec<S>) -> anyhow::Result<()> {
         let key = key.encode_key()?;
+
+        #[cfg(feature = "std")]
         self.rows
             .lock()
             .expect("Lock must not be poisoned")
             .entry(S::COLUMN_FAMILY_NAME)
             .or_default()
             .push(WriteOp::Deletion { key });
 
+        #[cfg(not(feature = "std"))]
+        self.rows
+            .lock()
+            .entry(S::COLUMN_FAMILY_NAME)
+            .or_default()
+            .push(WriteOp::Deletion { key });
+
         Ok(())
     }
 }
 
 /// An error that occurred during (de)serialization of a [`Schema`]'s keys or
 /// values.
-#[derive(Error, Debug)]
+#[derive(Debug)]
+#[cfg_attr(feature = "std", derive(thiserror::Error))]
 pub enum CodecError {
     /// Unable to deserialize a key because it has a different length than
     /// expected.
-    #[error("Invalid key length. Expected {expected:}, got {got:}")]
+    #[cfg_attr(
+        feature = "std",
+        error("Invalid key length. Expected {expected:}, got {got:}")
+    )]
     #[allow(missing_docs)] // The fields' names are self-explanatory.
     InvalidKeyLength { expected: usize, got: usize },
     /// Some other error occurred when (de)serializing a key or value. Inspect
     /// the inner [`anyhow::Error`] for more details.
-    #[error(transparent)]
-    Wrapped(#[from] anyhow::Error),
+    #[cfg_attr(feature = "std", error(transparent))]
+    Wrapped(#[cfg_attr(feature = "std", from)] anyhow::Error),
     /// I/O error.
-    #[error(transparent)]
-    Io(#[from] std::io::Error),
+    #[cfg_attr(feature = "std", error(transparent))]
+    Io(#[cfg_attr(feature = "std", from)] io::Error),
 }
 
-/// For now we always use synchronous writes. This makes sure that once the operation returns
-/// `Ok(())` the data is persisted even if the machine crashes. In the future we might consider
-/// selectively turning this off for some non-critical writes to improve performance.
-fn default_write_options() -> rocksdb::WriteOptions {
-    let mut opts = rocksdb::WriteOptions::default();
-    opts.set_sync(true);
-    opts
+#[cfg(not(feature = "std"))]
```

Comment:
Let's move all the conversions to a separate module and feature gate only the module.

Reply:
Ditto; let's centralize on the previous comment